docs/dart/testing.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Testing
+sidebar_position: 6
+---
+
+Testing a client application can be a crucial part of ensuring its functionality
+and performance. When it comes to mobile/web applications, spinning up a full server
+to test against may not always be the best option. In the following sections, we
+will go through a couple of alternatives.
+
+## Widget Testing
+
+For widgets that interface with Connect, it will generally be desirable to mock your RPCs since a backend may not be
```

Comment:
```suggestion
For widgets that interface with Connect, it will generally be preferable to mock your RPCs since a backend may not be
```

This is super-nitty, but just figured using a different word sounds better since `desirable` is used twice here.